timesketch/frontend-ng/src/views/Explore.vue

```diff
@@ -46,274 +44,251 @@ limitations under the License.
-              <v-chip-group>
-                <span v-for="(chip, index) in filterChips" :key="index + chip.value">
-                  <v-chip>
-                    {{ chip.value | formatLabelText }}
-                  </v-chip>
-                  <v-btn v-if="index + 1 < timeFilterChips.length" icon small style="margin-top: 2px" class="mr-2"
-                    >AND</v-btn
+              </v-chip>
+            </template>
+            <v-card>
+              <v-list>
+                <v-list-item>
+                  <v-list-item-action>
+                    <v-icon>mdi-square-edit-outline</v-icon>
+                  </v-list-item-action>
+                  <v-list-item-subtitle>Edit filter</v-list-item-subtitle>
+                </v-list-item>
+
+                <v-list-item @click="toggleChip(chip)">
+                  <v-list-item-action>
+                    <v-icon v-if="chip.active">mdi-eye-off</v-icon>
+                    <v-icon v-else>mdi-eye</v-icon>
+                  </v-list-item-action>
+                  <v-list-item-subtitle
+                    ><span v-if="chip.active">Temporarily disable</span
+                    ><span v-else>Re-enable</span></v-list-item-subtitle
                   >
-                </span>
-              </v-chip-group>
-            </v-col>
-          </v-row>
-        </v-card>
-      </v-col>
-    </v-row>
+                </v-list-item>
+                <v-list-item @click="removeChip(chip)">
+                  <v-list-item-action>
+                    <v-icon>mdi-delete</v-icon>
+                  </v-list-item-action>
+                  <v-list-item-subtitle>Remove filter</v-list-item-subtitle>
+                </v-list-item>
+              </v-list>
+            </v-card>
+          </v-menu>
+          <v-btn v-if="index + 1 < timeFilterChips.length" icon small style="margin-top: 2px" class="mr-2">OR</v-btn>
+        </span>
+        <span>
+          <v-menu
+            v-model="timeFilterMenu"
+            offset-y
+            :close-on-content-click="false"
+            :close-on-click="true"
+            content-class="menu-with-gap"
+            allow-overflow
+            style="overflow: visible"
+          >
+            <template v-slot:activator="{ on, attrs }">
+              <v-chip outlined v-bind="attrs" v-on="on">
+                <v-icon left small> mdi-clock-plus-outline </v-icon>
+                Add timefilter
+              </v-chip>
+            </template>
 
-    <!-- Search History -->
-    <v-row>
-      <v-col v-show="showSearchHistory" cols="12">
-        <v-card outlined>
-          <v-toolbar elevation="0" dense>
-            <v-toolbar-title>Search history</v-toolbar-title>
-            <v-spacer></v-spacer>
+            <ts-filter-menu app @cancel="timeFilterMenu = false" @addChip="addChip"></ts-filter-menu>
+          </v-menu>
+        </span>
+      </v-chip-group>
+
+      <!-- TODO: Move this control to the timeline picker -->
```

Comment:
Can we create an Issue for this TODO?

Reply:
Good point, issue created and comment updated.